Gems/Atom/RHI/WebGPU/Code/Source/RHI/Conversions.cpp

```diff
@@ -107,7 +107,49 @@ namespace AZ::WebGPU
             RHIWGPU_EXPAND_FOR_FORMATS(RHIWGPU_WGPU_TO_RHI)
 
         default:
-            AZ_Assert(false, "unhandled conversion in ConvertFormat");
+            AZ_Assert(false, "unhandled conversion in ConvertImageFormat");
+            return RHI::Format::Unknown;
+        }
+
+#undef RHIWGPU_WGPU_TO_RHI
+    }
+#undef RHIWGPU_EXPAND_FOR_FORMATS
+
+    // definition of the macro "RHIWGPU_EXPAND_FOR_FORMATS" containing formats' names.
+    #include "VertexFormats.inl"
+
+    wgpu::VertexFormat ConvertVertexFormat(RHI::Format format)
+    {
+#define RHIWGPU_RHI_TO_WGPU(_FormatID, _WGPUFormat)                                                              \
+    case RHI::Format::_FormatID:                                                                                 \
+        return wgpu::VertexFormat::_WGPUFormat;
```

Comment:
Just to double check that the return is intended, I might read it wrong but seems like you got unreachable code below

Reply:
This is fine. Its just a macro that gets used in the method below.